modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java

```diff
@@ -444,6 +446,39 @@ public void testGetSegmentsEntryIdsWithSingleModelCriterion()
 				segmentsEntryIds));
 	}
 
+	@Test
+	public void testGetSegmentsEntryIdsWithUserModelCriterionAndUserCreatedAfterSegmentsEntry()
+		throws Exception {
+
+		Criteria criteria = new Criteria();
+
+		String firstName = RandomTestUtil.randomString();
+
+		_userSegmentsCriteriaContributor.contribute(
+			criteria, String.format("(firstName eq '%s')", firstName),
+			Criteria.Conjunction.AND);
+
+		SegmentsEntry segmentsEntry = SegmentsTestUtil.addSegmentsEntry(
+			_group.getGroupId(), CriteriaSerializer.serialize(criteria),
+			User.class.getName());
+
+		_user1 = UserTestUtil.addUser();
```

Comment:
could we create the user with the right name in only one step?

Reply:
Good point, improving the user creating..